docs/source/howto/workchains_restart.rst

```diff
@@ -0,0 +1,402 @@
+
+.. figure:: include/images/workflow_error_handling_basic_failed.png
+
+    Example execution of the Quantum ESPRESSO phonon workflow where the third step, the ``q2r.x`` code, failed, and because the workflow blindly assumed it would have finished without errors also fails.
+
+The solution seems simple then.
+After each calculation, we simply add a check to verify that it finished successfully and produced the required outputs before continuing with the next calculation.
+What do we do, though, when the calculation failed?
+Depending on the cause of the failure, we might actually be able to fix the problem, and re-run the calculation, potentially with corrected inputs.
+A common example is that the calculation ran out of wall-time (requested time from the job scheduler) and was cancelled by the job scheduler.
+In this case, simply restarting the calculation (if the code supports restarts), and optionally giving the job more wall-time or resources, may fix the problem.
+
+You might be tempted to add this error handling directly into the workflow.
+However, you will find yourself implementing the same error-handling code many times in other workflows that just happen to have to run the same codes.
+For example, we could add the error-handling for the ``pw.x`` code directly in our phonon workflow, but a structure optimization workflow will also have to run ``pw.x`` and will have to implement the same error-handling logic.
+Is there a way that we can implement this once and easily reuse it in various workflows?
+
+Yes! Instead of directly running a calculation in a workflow, one should rather run a work chain that is explicitly designed to run the calculation to completion.
+This *base* work chain knows about the various failure modes of the calculation and can try to fix the problem and restart the calculation whenever it fails, until it finishes successfully.
+This logic of such a base work chain is very generic and can be applied to any calculation, and actually any process:
+
+.. figure:: include/images/workflow_error_handling_flow_base.png
+    :align: center
+    :height: 500px
+
+    Schematic flow diagram of the logic of a *base* work chain, whose job it is to run a subprocess repeatedly, fixing any potential errors, until it finishes successfully.
+
+The work chain runs the subprocess.
+Once it has finished, it then inspects the status.
+If the subprocess finished successfully, the work chain returns the results and its job is done.
+If, instead, the subprocess failed, the work chain should inspect the cause of failure, and attempt to fix the problem and restart the subprocess.
+This cycle is repeated until the subprocess finishes successfully.
+Of course this runs the risk of entering into an infinite loop if the work chain never manages to fix the problem, so we want to build in a limit to the maximum number of calculations that can be re-run:
+
+.. _workflow-error-handling-flow-loop:
```

Comment:
Since the tutorial, @sphuber has actually updated the schematic for the `PwBaseRestartWorkChain`:

![base_restart](https://user-images.githubusercontent.com/25607960/108207910-a4a77300-7128-11eb-8d0b-7de9066916fa.png)

This more accurately represents the steps taken, also including e.g. the fact that a process with an unhandled error will be run again if it's not the second consecutive unhandled error.

Reply:
changed

Reply:
Whoops! 😅  I accidentally added the version I made to get bigger fonts for a report. Here is Seb's version, that has the same style as the other images:

![flow_base_restart](https://user-images.githubusercontent.com/25607960/108223011-a168b300-7139-11eb-9ae2-c718614f5ac3.png)

Reply:
you fooooool 🤡 . alright added

Reply:
can you approve 😉